docs/references/cluster-metrics.md

```diff
@@ -0,0 +1,185 @@
+You can use these templates as references to build your own dashboards.
+For any metrics that are missing in the dashboards, use [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) as a reference.
+
+Note that, apart from these metrics emitted by the Cluster, you should also monitor infrastructure-specific metrics like CPU, memory, and network for all hosts that are running Temporal Cluster services.
+
+## Common metrics
+
+Temporal emits metrics for each gRPC service request.
+These metrics are emitted with `type`, `operation`, and `namespace` tags, which provide visibility into Service usage and show the request rates across Services, Namespaces, and Operations.
+
+- Use the `operation` tag in your query to get request rates, error rates, or latencies per operation.
+- Use the `service_name` tag with the [service role tag values](https://github.com/temporalio/temporal/blob/bba148cf1e1642fd39fa0174423b183d5fc62d95/common/metrics/defs.go#L108) to get details for the specific service.
+
+All common tags that you can add in your query are defined in the following file:
+https://github.com/temporalio/temporal/blob/86966c5d2f78ee74c10b6808857f1a2f64d3c134/common/metrics/defs.go#L90.
+
+For example, to see service requests by operation on the Frontend Service, use the following:
+
+`sum by (operation) (rate(service_requests{service_name="frontend"}[2m]))`
+
+Note: All metrics queries in this topic are [Prometheus queries](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+
+The following list describes some metrics you can get started with.
+
+### `service_requests`
+
+Shows service requests received per Task Queue.
+Example: Service requests by operation
+`sum(rate(service_requests{operation=\"AddWorkflowTask\"}[2m]))`
+
+### `service_latency`
+
+Shows latencies for all Client request operations.
+Usually these are the starting point to investigate which operation is experiencing high-latency issues.
+Example: P95 service latency by operation for the Frontend Service
+`histogram_quantile(0.95, sum(rate(service_latency_bucket{service_name="frontend"}[5m])) by (operation, le))`
+
+### `service_error_with_type`
+
+(Available only in v1.17.0+) Identifies errors encountered by the service.
+Example: Service errors by type for the Frontend Service
+`sum(rate(service_errors_with_type{service_name="frontend"}[5m])) by (error_type)`
+
+### `client_errors`
+
+An indicator for connection issues between different Server roles.
+Example: Client errors
+`sum(rate(client_errors{service_name="frontend",service_role="history"}[5m]))`
+
+In addition to these, you can define some service-specific metrics to get performance details for each service.
+Start with the following list, and use [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) to define additional metrics as required.
+
+## Matching Service metrics
```

Comment:
One interesting metric from matching is no_poller_tasks, it is a counter metric that is emitted whenever a task is added to a task queue that has no poller. This usually indicate wrong task queue is used either by the starter or by the worker.